benchexec/runexecutor.py

```diff
@@ -515,14 +517,32 @@ def _setup_environment(self, environments):
         if environments.get("keepEnv", None) is not None:
             run_environment = {}
         else:
-            run_environment = os.environ.copy()
+            run_environment = {key: val for key, val in os.environ.items()}
         for key in environments.get("keepEnv", {}).keys():
             if key in os.environ:
                 run_environment[key] = os.environ[key]
         for key, value in environments.get("newEnv", {}).items():
             run_environment[key] = value
         for key, value in environments.get("additionalEnv", {}).items():
             run_environment[key] = os.environ.get(key, "") + value
+        if environments.get("randomizeEnvSize", None) is not None:
+            # Caller should consider setting random.seed(...) for improved
+            # (but not complete) determinism in the runtime.
```

Comment:
This comment with a note for callers seems a little bit out-of-place, because developers of callers won't see it here.

Also I don't really understand the "should" here: Isn't the whole point of this feature the non-determinism? Why should one want to fix a seed? And if this should indeed be done, why can't we do this for the caller? If we don't do this, callers will just forget it, and callers of the command-line `runexec` will not have the chance to do this anyway.